app/src/org/commcare/dalvik/geo/HereFunctionHandler.java

```diff
@@ -101,27 +99,27 @@ public void onLocationChanged(Location location) {
         }
     }
 
+    public boolean locationProvidersFound() {
+        return GeoUtils.evaluateProvidersWithPermissions(mLocationManager, context).size() > 0;
+    }
```

Comment:
Since you always invert the value returned by this method I would rename it to `areLocationProvidersMissing` and use `.isEmpty` instead of `.size() == 0`